src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Security/PermissionExtensions.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using OrchardCore.Security.Permissions;
+
+namespace OrchardCore.Security
+{
+    public static class PermissionExtensions
+    {
+        /// <summary>
+        /// Evaluates if the specified <see cref="Permission"/> is granted by provided claims.
+        /// </summary>
+        /// <param name="permission">The <see cref="Permission"/> to test</param>
+        /// <param name="claims">Provided claims.</param>
+        /// <returns>True if the permission is granted, otherwise false.</returns>
+        public static bool IsGranted(this Permission permission, IEnumerable<Claim> claims)
```

Comment:
We might want to introduce this as a service - @sebastienros would confirm, but last we spoke about it, the implied by's were showing up on stack traces a lot, and with a service (even if not down now on this pr), we would have the potential to add some caching to the implied by `HashSet`, rather than having to change this api again.

Reply:
An alternative could be to add "Anonymous" permissions using a middleware, and "Authenticated" permissions using an `IUserClaimsProvider`, so all authorized permissions will be available as claims.
Then we can get rid of `RolesPermissionsHandler`.

In that case, the `IsGranted` logic will go into `PermissionHandler` - which is already a service - and we could add any required cache logic.